Realm/RLMAccessor.hpp

```diff
@@ -0,0 +1,116 @@
+////////////////////////////////////////////////////////////////////////////
+//
+// Copyright 2017 Realm Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+////////////////////////////////////////////////////////////////////////////
+
+#import "RLMAccessor.h"
+
+#import "object_accessor.hpp"
+
+#import "RLMUtil.hpp"
+
+@class RLMRealm;
+class RLMClassInfo;
+class RLMObservationInfo;
+
+// realm::util::Optional<id> doesn't work due to that explicitly invoking the
+// destructor isn't allowed under ARC, so this covers the subset of Optional
```

Comment:
Yes. I was mislead by the fact that the thing it happens to complain about is the destructor call, but that's just because that's the first time the value in the union is used and it complains on use rather than when the union is defined.